cmd/kodelet/pr_respond.go

```diff
@@ -150,48 +169,149 @@ func getPRRespondConfigFromFlags(cmd *cobra.Command) *PRRespondConfig {
+func fetchFocusedCommentData(prURL, commentID string) (string, string, error) {
+	// Extract repository information from PR URL
+	// Expected format: https://github.com/owner/repo/pull/number
+	parts := strings.Split(prURL, "/")
+	if len(parts) < 5 {
+		return "", "", fmt.Errorf("invalid PR URL format")
+	}
+	
+	owner := parts[3]
+	repo := parts[4]
+	
+	// Fetch the specific comment using GitHub API
+	cmd := exec.Command("gh", "api", fmt.Sprintf("repos/%s/%s/pulls/comments/%s", owner, repo, commentID),
+		"--jq", "{author: .user.login, body: .body, path: .path, line: .line, created_at: .created_at}")
+	commentOutput, err := cmd.Output()
+	if err != nil {
+		return "", "", fmt.Errorf("failed to fetch comment details: %w", err)
+	}
+	
+	focusedComment := fmt.Sprintf("Comment ID %s:\n%s", commentID, strings.TrimSpace(string(commentOutput)))
+	
+	// For related discussions, we can fetch all comments on the same file/line
+	// This is a simplified approach - in practice, you might want more sophisticated logic
+	cmd = exec.Command("gh", "api", fmt.Sprintf("repos/%s/%s/pulls/comments", owner, repo),
+		"--jq", fmt.Sprintf(".[] | select(.path == (.[] | select(.id == %s) | .path)) | {id: .id, author: .user.login, body: .body, line: .line, created_at: .created_at}", commentID))
+	discussionOutput, err := cmd.Output()
+	if err != nil {
+		// If we can't get related discussions, just return empty
+		relatedDiscussion := "No related discussions found or failed to fetch"
+		return focusedComment, relatedDiscussion, nil
+	}
+	
+	relatedDiscussion := fmt.Sprintf("Related discussions for comment %s:\n%s", commentID, strings.TrimSpace(string(discussionOutput)))
+	
+	return focusedComment, relatedDiscussion, nil
+}
+
+func generatePRRespondPrompt(bin, prURL, commentID string, prData *PRData) string {
 	commentInstruction := ""
+	focusedSections := ""
+	
 	if commentID != "" {
 		commentInstruction = fmt.Sprintf(`
-3. Focus on the specific comment ID: %s
-   - Use appropriate gh commands to retrieve the specific comment details
-   - Parse the comment content and understand the specific request`, commentID)
+
+Focus on the specific comment ID: %s by reviewing the focused comment and related discussions below.`, commentID)
+		
+		// Add focused comment and related discussions sections when comment ID is specified
+		focusedSections = fmt.Sprintf(`
+
+<pr_comment>
+%s
+</pr_comment>
+
+<pr_discussions>
+%s
+</pr_discussions>`, prData.FocusedComment, prData.RelatedDiscussion)
 	} else {
 		commentInstruction = `
-3. Find the most recent @kodelet mention:
-   - Look through all comments to find the latest mention of @kodelet
-   - Focus on that specific comment and its context
-   - If no @kodelet mention is found, address the most recent review comment`
+
+Find the most recent @kodelet mention by reviewing the comments data above. If no @kodelet mention is found, address the most recent review comment.`
```

Comment:
please use the same format as `focusedSections` above. You might need to manually find the latest `@kodelet` comment